src/tiled/snaphelper.cpp

```diff
@@ -55,6 +58,8 @@ void SnapHelper::snap(QPointF &pixelPos) const
             tileCoords = tileCoords.toPoint();
         }
         pixelPos = mRenderer->tileToPixelCoords(tileCoords);
+    } else if (mSnapToPixels) {
+        pixelPos = pixelPos.toPoint();
```

Comment:
I think the problem with this is, that even though internally this is called the "pixel position", it is not actually the "screen pixels" that the user would expect "snap to pixel" to work on. I think you'll need to do:
```c++
QPointF screenPos = mRenderer->pixelToScreenCoords(pixelPos);
pixelPos = mRenderer->screenToPixelCoords(screenPos.toPoint());
```